app/validation/validators.py

```diff
@@ -218,6 +219,16 @@ def __call__(self, form, field):
             raise validators.StopValidation(self.message)
 
 
+class YearCheck:
```

Comment:
Should the check be done as part of `DateCheck`, because I don't think a DateCheck vs YearCheck is very clear?.